src/rastervision/registry.py

```diff
@@ -95,6 +99,18 @@ def __init__(self):
             rv.AWS_BATCH: rv.runner.LocalExperimentRunner
         }
 
+        self.filesystems = [
+            (1, rfs.HttpFileSystem),
+            (2, rfs.S3FileSystem),
+            (2**64-1, rfs.LocalFileSystem)
```

Comment:
This is necessary if there is more than one filesystem that can potentially be used for a given uri.

Reply:
This is not necessary, though I see why it was put in place. However, the list order we check the filesystems will determine what gets used for a URI - so if multiple filesystems can work for a URI, whatever is higher on the list will work. The list order is controlled by us so there's no problem there ordering things correctly. As it currently stands the priority is essentially a call to `enumerate(self.filesystems)`.

For plugins, there's no way for a plugin author to know that their filesystem ranking will be higher than another plugin, or else the conflicting filesystems will be both registered by the single author, in which case they can register them in the order that they should be checked. If there are multiple plugins that can handle a given URI that conflict, that's a deeper edge case.

Given that, I think the priority is extraneous and potentially confusing to a plugin author (like I mentioned, my guess is that everyone will register things with priority 1, otherwise they can just register their filesystems in the order they want them searched). I'd vote for removal.

Reply:
Done.